src/term-spec.js

```diff
@@ -128,21 +128,133 @@ declare export class AssignmentTargetPropertyProperty extends AssignmentTargetPr
+  elements: (ClassElement | IndexSignatureDeclaration)[];
 }
 
 declare export class ClassElement extends Term {
-  isStatic: any;
+  isStatic: any; // boolean
+  accessModifier: any; // 'public' | 'protected' | 'private'
+}
+
+declare export class MethodClassElement extends ClassElement {
+  isAbstract: any; // boolean
+  hasQuestionToken: any; // boolean
   method: MethodDefinition;
 }
 
+declare export class ConstructorClassElement extends ClassElement {
+  params: ConstructorFormalParameters;
+  body: FunctionBody;
+}
+
+declare export class ConstructorFormalParameters extends Term {
+  items: ConstructorParameterDeclaration[];
+  rest?: ParameterDeclaration;
+}
+
+declare export class ConstructorParameterDeclaration extends Term {
+  accessModifier?: any; // 'public' | 'protected' | 'private'
+  hasReadonlyModifier: any; // boolean
+  parameter: ParameterDeclaration;
+}
+
+declare export class SemicolonClassElement extends ClassElement {}
+
+declare export class PropertyDeclaration extends ClassElement {
+  hasQuestionToken: any; // boolean
+  name: PropertyName;
+  type?: TypeNode;
+  initializer?: Expression;
+}
+
+
+
+// interface
+
+declare export class InterfaceDeclaration extends Statement {
+  name: BindingIdentifier;
+  typeParameters?: TypeParameterDeclaration[];
+  heritageClauses?: HeritageClause[];
+  elements: (TypeElement | IndexSignatureDeclaration)[];
+}
+
+declare export class TypeElement extends Term {}
+
+declare export class FunctionSignatureLikeTypeElement extends TypeElement {
+  typeParameters?: TypeParameterDeclaration[];
+  params: FormalParameters;
+  type?: TypeNode;
+}
+
+declare export class MethodSignature extends FunctionSignatureLikeTypeElement {
+  name: PropertyName;
+}
+
+declare export class CallSignatureDeclaration extends FunctionSignatureLikeTypeElement {}
+
+declare export class ConstructorSignatureDeclaration extends FunctionSignatureLikeTypeElement {
+  // note: TypeScript does not allow ConstructorFormalParameters here
+}
+
+declare export class PropertySignature extends TypeElement {
+  name: PropertyName;
+  hasQuestionToken: any; // boolean
+  type?: TypeNode;
+}
+
+
+
+// common for classes and interfaces
+
+declare export class HeritageClause extends Term {
+  parent?: ClassExpression | ClassDeclaration | InterfaceDeclaration;
+  types: ExpressionWithTypeArguments[];
+}
+declare export class ExtendsClause extends HeritageClause {}
+declare export class ImplementsClause extends HeritageClause {}
+
+declare export class IndexSignatureDeclaration extends Term {
+  parent?: ClassExpression | ClassDeclaration | InterfaceDeclaration | TypeLiteralNode;
+  parameter: ParameterDeclaration;
+  valueType: TypeNode;
+}
+
+
+
+// type alias
+
+declare export class TypeAliasDeclaration extends Statement {
+  name: BindingIdentifier;
+  typeParameters?: TypeParameterDeclaration[];
+  type: TypeNode;
+}
+
+
+
+// enum
+
+declare export class EnumDeclaration extends Statement {
```

Comment:
Enum support is not strictly related to type annotations, so they could be dropped from here.